server/apps/main/views.py

```diff
@@ -322,22 +326,43 @@ def moderation_list(request):
     else:
         return redirect("main:overview")
 
-# @vcr.use_cassette('tmp/my_stories.yaml', filter_query_parameters=['access_token'])
+
 def my_stories(request):
     """
     List all stories that are associated with the OpenHumans project page.
     Including those which are not-shareable on the website
     """
     if request.user.is_authenticated:
-        context = {"files": request.user.openhumansmember.list_files()}
+        files = request.user.openhumansmember.list_files()
+        context = {"files": files}
         context = reformat_date_string(context)
-        statuses = get_review_status(context["files"])
-        context = {**context, **statuses}
+
+        # Define the number of items per page
+        items_per_page = 10
+
+        # For each category, create a Paginator and get the page requested by the user
+        paginator_public = Paginator(filter_by_tag(filter_by_moderation_status(files, "approved"), "public"), items_per_page)
+        page_public = request.GET.get('page_public')
+        context['public_stories'] = paginator_public.get_page(page_public)
```

Comment:
This is really nice, but currently the pagination style doesn't quite match up with the pagination style on the moderation pages. In particular, it seems to be missing the first/last and ellipses entries. I.e. what I'd expect to see is something like this:

<img width="343" alt="image" src="https://github.com/alan-turing-institute/AutSPACEs/assets/1446122/036ce18c-3ee4-4116-ae8a-0218e27b1ab2">

Whereas in practice I'm seeing this (no ellipses, no first entry):

<img width="288" alt="image" src="https://github.com/alan-turing-institute/AutSPACEs/assets/1446122/c774b850-882e-4eb6-93be-3ed515b69e23">

It would be nice if these could be added in I think. It should be possible to use [`Paginator.get_elided_page_range()`](https://docs.djangoproject.com/en/4.2/ref/paginator/#django.core.paginator.Paginator.get_elided_page_range) for this.

Reply:
Yes, again good point and thanks for hint to` get_elided_page_range()`. Changed that.